lualib/rspamadm/secretbox.lua

```diff
@@ -0,0 +1,101 @@
+local util = require 'lua_util'
+local rspamd_util = require 'rspamd_util'
+local argparse = require 'argparse'
+
+local parser = argparse()
+    :name "rspamadm encrypt/decrypt text"
+    :description "Encrypt/Decrypt given text with given key and nonce"
+    :help_description_margin(32)
+    :command_target('command')
+    :require_command(true)
+
+local decrypt = parser:command 'decrypt'
+                      :description 'Decrypt text with given key and nonce'
+
+decrypt:option "-t --text"
+      :description("Encrypted text(Base 64)")
+      :argname("<text>")
+decrypt:option "-k --key"
+      :description("Key used to encrypt text")
+      :argname("<key>")
+decrypt:option "-n --nonce"
+      :description("Nonce used to encrypt text(Base 64)")
+      :argname("<nonce>")
+      :default('nil')
+
+local encrypt = parser:command 'encrypt'
+                      :description 'Encrypt text with given key'
+
+encrypt:option "-t --text"
+      :description("Text to encrypt")
+      :argname("<text>")
+encrypt:option "-k --key"
+      :description("Key to encrypt text")
+      :argname("<key>")
+encrypt:option "-n --nonce"
+       :description("Nonce to encrypt text(Base 64)")
+       :argname("<nonce>")
+
+local function decryption_handler(args)
+    local settings = {
+        prefix = 'dec',
+        dec_encrypt = true,
+        dec_key = args.key
+    }
+    local decrypted_header = ''
+    if(args.nonce ~= 'nil') then
+         decrypted_header = util.maybe_decrypt_header(rspamd_util.decode_base64(args.text), settings, settings.prefix,
```

Comment:
Hold on, why is it `base64`?? We should be flexible about encoding: raw, base64, base32, hex etc. Please look how other utilities are organized.

Reply:
I decided to do it as a mutex of flags. I thing it will be pretty convenient to user to manage encodings in this way.